File: day2/day2.R
#https://adventofcode.com/2021/day/2

##PART 1
inp <- read.csv(here::here("day2/input.csv"), header = FALSE)

x <- unlist(inp)

nms <- gsub(" \\d+", "", x)
vals <- as.numeric(gsub("\\D+", "", x))

names(vals) <- nms

y <- sum(vals[names(vals) == "forward"]) * (sum(vals[names(vals) == "down"]) - sum(vals[names(vals) == "up"]))

##PART 2
library(tidyverse)

#sum of forward will remain the same
f <- sum(vals[names(vals) == "forward"])

df <- tibble(
  cmd = names(vals),
  val = vals
) %>%
  mutate(aim_delta = case_when(
    cmd == "forward" ~ 0,
    cmd == "down" ~ vals,
    cmd == "up" ~ -1*vals
  ),
  aim_cur = cumsum(aim_delta))

depth <- df %>% 
  filter(cmd == "forward") %>%
  mutate(depth = aim_cur*val) %>%
  summarize(sum(depth)) %>%
  pull()

f*depth

File: day2/day2.jl
using Revise
using CSV
using DataFrames

inp = CSV.read("day2/input.csv", DataFrame, header = 0)

## part 1

frwd = sum(filter(:Column1 => ==("forward"), inp)[:,2])
up = sum(filter(:Column1 => ==("up"), inp)[:,2])
down = sum(filter(:Column1 => ==("down"), inp)[:, 2])

res1 = frwd * (down - up)

res1

## part 2

aim = 0
depth = 0
horz = 0

for i in 1:nrow(inp)

    if inp[i, 1] == "forward"
        global horz += inp[i, 2]
        global depth += aim * inp[i, 2]
    elseif inp[i, 1] == "up"
        global aim -= inp[i, 2]
    else inp[i, 1] == "down"
        global aim += inp[i, 2]
    end
end

res2 = depth * horz

res2
    